api/all/src/main/java/io/opentelemetry/api/trace/TraceFlags.java

```diff
@@ -8,47 +8,128 @@
 import javax.annotation.concurrent.Immutable;
 
 /**
- * Helper methods for dealing with trace flags options. These options are propagated to all child
- * {@link Span spans}. These determine features such as whether a {@code Span} should be traced. It
- * is implemented as a bitmask.
+ * Helper methods for dealing with trace flags options. A valid trace flags is a 2 character
+ * lowercase hex (base16) String.
+ *
+ * <p>These options are propagated to all child {@link Span spans}. These determine features such as
+ * whether a {@code Span} should be traced. It is implemented as a bitmask.
  */
 @Immutable
 public final class TraceFlags {
```

Comment:
@trask Will this prevent the interface bridging of `SpanContext` being a class? I guess so since we wouldn't be able to use the covariant return type.

@bogdandrutu Can we make this an interface and move everything into `ImmutableTraceFlags` class?

Reply:
Don't see a problem with that

Reply:
Can I do it in the next PR to limit the size? Or I can do it here if you really need it atomically.

Reply:
Nah follow up is fine